contracts/token/psp22/traits.rs

```diff
@@ -353,3 +372,74 @@ pub trait PSP22Receiver {
         data: Vec<u8>,
     ) -> Result<(), PSP22ReceiverError>;
 }
+
+#[brush::trait_definition]
+pub trait PSP22Mintable: PSP22 {
+    #[ink(message)]
+    fn mint(&mut self, amount: Balance) {
+        self._mint(Self::env().caller(), amount);
+    }
+}
+
+/// Extension of [`PSP22`] that allows token holders to destroy both their own
+/// tokens and those that they have an allowance for, in a way that can be
+/// recognized off-chain (via event analysis).
+#[brush::trait_definition]
+pub trait PSP22Burnable: PSP22 {
+    /// Destroys `amount` tokens from the caller.
+    ///
+    /// See [`PSP22::_burn`].
+    #[ink(message)]
+    fn burn(&mut self, amount: Balance) {
+        self._burn(Self::env().caller(), amount);
+    }
+
+    /// Destroys `amount` tokens from `account`, deducting from the caller's
+    /// allowance.
+    ///
+    /// See [`PSP22::_burn`] and [`PSP22::allowance`].
+    ///
+    /// Requirements:
+    ///
+    /// - the caller must have allowance for ``accounts``'s tokens of at least
+    /// `amount`.
+    /// # Errors
+    ///
+    /// Panics with `InsufficientAllowance` error if there are not enough tokens allowed
+    /// by owner for `spender`.
+    #[ink(message)]
+    fn burn_from(&mut self, account: AccountId, amount: Balance) {
+        let current_allowance = self.allowance(account, Self::env().caller());
```

Comment:
You can use `_allowance` and in this case, you don't need to copy arguments

Reply:
I only see allowance. Do you mean use self.allowance(..)?